### PR TITLE
Feature 176 - LINQ queries

### PR DESCRIPTION
**Pull request by AVLozhechkin**
PBI #176 
Note: Source files I worked with are formatted according to the StyleCop.

### CODE REVIEW

File: HW5/Artem_Lozhechkin/Task/Task/App.config [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+    <configSections>
+    </configSections>
+    <connectionStrings>
+        <add name="Task.Properties.Settings.NorthwindConnectionString"
+            connectionString="Data Source=srv-fund2019.database.windows.net;Initial Catalog=Northwind;User ID=xadmin;Password=P@ssw0rd"
```
Comment (MihailRom) [inline HW5/Artem_Lozhechkin/Task/Task/App.config]: I think it’s very bad to upload a file with a username and password to the repository.

File: HW5/Artem_Lozhechkin/Task/Task/LinqSamples.cs [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+        {
+            var products = this.dataSource.Products
+                .GroupBy(p => p.Category)
+                .Select(c => new
+                {
+                    Category = c.Key,
+                    Stocks = c.GroupBy(s => s.UnitsInStock)
+                .Select(e => new
+                {
+                    Left = e.Key == 0 ? "Sold" : $"{e.Key} species left",
```
Comment (MihailRom) [inline HW5/Artem_Lozhechkin/Task/Task/LinqSamples.cs]: Perhaps, the sorting by availability implied 2 groups - is it in stock or not. And inside them is already sorted by cost.

File: HW5/Artem_Lozhechkin/Task/Task/LinqSamples.cs [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+                    Prices = e
+                .Select(pr => new { Name = pr.ProductName, Price = pr.UnitPrice }).OrderBy(pr => pr.Price)
```
Comment (Foxtra) [inline HW5/Artem_Lozhechkin/Task/Task/LinqSamples.cs]: Looks like there is wrong sorting by price
изображение

File: HW5/Artem_Lozhechkin/Task/Task/LinqSamples.cs [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        public void LinqTask1()
+        {
+            int x = 20000;
+
+            // int x = 15000;
+            // int x = 10000;
+            var customers =
+                from c in this.dataSource.Customers
+                where c.Orders.Select(price => price.Total).Sum() > x
+                select c.CompanyName;
```
Comment (Foxtra) [inline HW5/Artem_Lozhechkin/Task/Task/LinqSamples.cs]: It is hard to check, when only the name of client in output of query.
For me would be better if the price field will be shown also, then I could exactly see that all outputed names corect.
`select new { c.CompanyName, price = c.Orders.Select(price => price.Total).Sum() };`

File: HW5/Artem_Lozhechkin/Task/Task/LinqSamples.cs (hunk 3 above)
Comment (AVLozhechkin) [reply HW5/Artem_Lozhechkin/Task/Task/LinqSamples.cs]: Yes, it was wrong grouping. I fixed it.

File: HW5/Artem_Lozhechkin/Task/Task/LinqSamples.cs (hunk 2 above)
Comment (AVLozhechkin) [reply HW5/Artem_Lozhechkin/Task/Task/LinqSamples.cs]: Yes, fixed it.

File: HW5/Artem_Lozhechkin/Task/Task/App.config (hunk 1 above)
Comment (AVLozhechkin) [reply HW5/Artem_Lozhechkin/Task/Task/App.config]: I thought that all repository's members know this log-pass, so I didn't hide it. But anyway I replaced it with a stars.

**Pull request merged**